add v2 wallet settings page

Sirmorrison opened the pull request:
This Pr Fix #1033 
This Pr implement the V2 dcr wallet settings design and all its functionalities.

<img width="1278" alt="image" src="https://user-images.githubusercontent.com/27733432/185481737-3ba4b62a-bff3-4711-b338-21958d346270.png">

Sirmorrison committed:
add v2 wallet settings page

Sirmorrison committed:
update general section features

Sirmorrison committed:
update security tools section

Sirmorrison committed:
add debug and danger zone functionalities

Sirmorrison committed:
add settings account functionality, clean up code

Sirmorrison committed:
localize string variables, fix new acct display bug

devchoplife commented:
Regarding the description, I think closes is a better verb because the issue is not a bug.

devchoplife commented on ui/page/security/validate_address.go:
```diff
@@ -32,8 +33,7 @@ type ValidateAddressPage struct {
 	stateValidate         int
-	walletName            string
-	isEnabled             bool
+	_                     string
```
This variable was declared but not used. Also, I am not sure why it was created.

devchoplife commented on ui/values/localizable/es.go:
```diff
@@ -12,7 +12,7 @@ const ES = `
 "online" = "En línea, ";
 "offline" = "Desconectado, ";
-"showDetails" = "Mostrar detalles";
+"viewDetails" = "View details";
```
This should be translated to spanish

dreacot replied:
it doesn't necessarily have to, we would have a translator do all non-english translations

Sirmorrison replied:
The idea is to have a place holder for the other languages pending when we do a general sting localization like the #906

devchoplife commented on ui/values/localizable/es.go:
```diff
@@ -90,7 +90,8 @@ const ES = `
 "stakeShuffle" = "StakeShuffle";
 "rename" = "Renombrar";
-"renameWalletSheetTitle" = "Renombrar la cartera";
+"renameWalletSheetTitle" = "Change wallet name";
+"securityToolsInfo" = "%v Various tools that help in different aspects of crypto currency security will be located here. %v"
```
This should be translated to spanish

Sirmorrison replied:
The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

devchoplife commented on ui/values/localizable/es.go:
```diff
@@ -478,4 +479,8 @@ syncCompTime" = "Est. Sync completion time"
 "selectDexServerToOpen" = "Select the Dex server you would like to open."
 "addDexServer" = "Add dex server"
 "canBuy" = "Can Buy"
+"removeWalletInfo" = "%v Are you sure you want to remove %v %s%v? Enter the name of the wallet below to verify. %v"
+"propNotif" = "Proposal notification"
+"peer" = "Peer"
+"confirmUmixedSpending" = "Confirm to allow spending from unmixed accounts"
```
This should be translated to spanish

dreacot replied:
it doesn't necessarily have to, we would have a translator do all non-english translations

Sirmorrison replied:
The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

devchoplife commented on ui/values/localizable/fr.go:
```diff
@@ -24,7 +24,7 @@ const FR = `
 "unlockWithPassword" = "Déverrouiller avec mot de passe"
 "offline" = "Hors ligne, ";
-"showDetails" = "Montrer les détails";
+"viewDetails" = "View details";
```
This should be translated to French

dreacot replied:
it doesn't necessarily have to, we would have a translator do all non-english translations

Sirmorrison replied:
The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

devchoplife commented on ui/values/localizable/fr.go:
```diff
@@ -50,7 +50,8 @@ const FR = `
 "timeLeft" = "%v restant";
 "currencyConversion" = "Conversion de devise";
-"renameWalletSheetTitle" = "Renommer le porte-monnaie";
+"renameWalletSheetTitle" = "Change wallet name";
+"securityToolsInfo" = "%v Various tools that help in different aspects of crypto currency security will be located here. %v"
```
This should be translated to French

dreacot replied:
it doesn't necessarily have to, we would have a translator do all non-english translations

Sirmorrison replied:
The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

devchoplife commented on ui/values/localizable/fr.go:
```diff
@@ -455,4 +456,8 @@ const FR = `
 "selectDexServerToOpen" = "Select the Dex server you would like to open."
 "addDexServer" = "Add dex server"
 "canBuy" = "Can Buy"
+"removeWalletInfo" = "%v Are you sure you want to remove %v %s%v? Enter the name of the wallet below to verify. %v"
+"propNotif" = "Proposal notification"
+"peer" = "Peer"
+"confirmUmixedSpending" = "Confirm to allow spending from unmixed accounts"
```
This should be translated to French

dreacot replied:
it doesn't necessarily have to, we would have a translator do all non-english translations

Sirmorrison replied:
The idea is to have a place holder for the other languages pending when we do a general sting localization like the https://github.com/planetdecred/godcr/pull/906

devchoplife reviewed (changes requested):
- The page content width should be reduced as seen on the mockup:
![image](https://user-images.githubusercontent.com/66803475/184950479-760140ba-965b-4026-84db-d4a675e88698.png)

vs 
![image](https://user-images.githubusercontent.com/66803475/184950585-4a3a2f4b-2c04-4cbc-b950-e53484cf4775.png)

dreacot reviewed (dismissed):
1. When disabling fetching of proposals, proposals notifications should also default to being turned off
![Screenshot from 2022-08-16 19-47-58](https://user-images.githubusercontent.com/25265396/184956292-78f06fff-ead8-4582-8d32-981aadcdf9cc.png)

2. clicking on "Check wallet log" crashes the app. see log below;

```
2022/08/16 19:49:05 FATAL -- failed to create Watcher
goroutine 322 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/nxadm/tail/util.Fatal({0x17e8c8b, 0xc8b90b}, {0x0, 0x0, 0x0})
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/util/util.go:22 +0x97
github.com/nxadm/tail/watch.(*InotifyTracker).run(0xc006e10600)
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:219 +0x68
created by github.com/nxadm/tail/watch.glob..func1
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:54 +0x173
exit status 1
```
3. The modal below should say "OK" based on the mockup, it also looks like there is excessive top and bottom spacing. on second thought, due to the vast amount of modals that requiring changing, this would be better suited in it's own PR
![Screenshot from 2022-08-16 19-52-39](https://user-images.githubusercontent.com/25265396/184957669-6aa89262-aec9-4827-bcac-4431e59fe7a9.png)
![Screenshot from 2022-08-16 19-53-47](https://user-images.githubusercontent.com/25265396/184957982-3966efa0-7ed2-481d-8ebb-0842a15ca957.png)

4.

Sirmorrison commented:
> Regarding the description, I think closes is a better verb because the issue is not a bug.

i don't really think it matters... the idea is to link the pr to the issue and it does just that..

Sirmorrison commented:
> * The page content width should be reduced as seen on the mockup:
>       ![image](https://user-images.githubusercontent.com/66803475/184950479-760140ba-965b-4026-84db-d4a675e88698.png)
> 
> 
> vs ![image](https://user-images.githubusercontent.com/66803475/184950585-4a3a2f4b-2c04-4cbc-b950-e53484cf4775.png)

the mockup shows the design in full screen... kindly open the app in full display and compare.

Sirmorrison committed:
Add page indentation as mockup.
- add ok string local and replace Got it strings on settings page.
- fix bug with proposal notification being active when proposal fetching is disabled
- minor code cleanup

Sirmorrison commented:
@dreacot i could not reproduce this. what where your repro steps? also the log does not point to any godcr file.

2022/08/16 19:49:05 FATAL -- failed to create Watcher
goroutine 322 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/nxadm/tail/util.Fatal({0x17e8c8b, 0xc8b90b}, {0x0, 0x0, 0x0})
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/util/util.go:22 +0x97
github.com/nxadm/tail/watch.(*InotifyTracker).run(0xc006e10600)
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:219 +0x68
created by github.com/nxadm/tail/watch.glob..func1
        /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:54 +0x173
exit status 1

devchoplife commented:
> @dreacot i could not reproduce this. what where your repro steps? also the log does not point to any godcr file.
> 
> 2022/08/16 19:49:05 FATAL -- failed to create Watcher goroutine 322 [running]: runtime/debug.Stack() /usr/local/go/src/runtime/debug/stack.go:24 +0x65 github.com/nxadm/tail/util.Fatal({0x17e8c8b, 0xc8b90b}, {0x0, 0x0, 0x0}) /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/util/util.go:22 +0x97 github.com/nxadm/tail/watch.(*InotifyTracker).run(0xc006e10600) /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:219 +0x68 created by github.com/nxadm/tail/watch.glob..func1 /home/kennedy/Projects/pkg/mod/github.com/nxadm/tail@v1.4.4/watch/inotify_tracker.go:54 +0x173 exit status 1

I cannot reproduce the error too

Sirmorrison committed:
implement new add acct functionality

JustinBeBoy commented:
1. add notice when sign message success like mockup and remove background of "signature" text
<img width="291" alt="image" src="https://user-images.githubusercontent.com/19331824/185579212-dff66e02-0d9b-47d5-8536-00503ed2e110.png">

2. "DCR" not "DC"
<img width="797" alt="aaaaaa" src="https://user-images.githubusercontent.com/19331824/185579475-c09bc0f6-9283-4bed-a379-2f56a7886fbe.png">

3. Seem like modal to create new account of wallet ui not like mockup
app:
<img width="898" alt="bb1" src="https://user-images.githubusercontent.com/19331824/185579758-abd2db3d-dc15-458a-b17c-3a55b9358b80.png">
vs
mockup:
<img width="371" alt="bb2" src="https://user-images.githubusercontent.com/19331824/185579848-a73f4db3-6f82-4e20-91c2-b408bf5f6d90.png">

4. modal info of verify message and sign message not like mockup
<img width="578" alt="cccc" src="https://user-images.githubusercontent.com/19331824/185580138-f7c692c9-7063-4615-bc28-32f8ba2d68b9.png">
vs 
<img width="259" alt="cccc2" src="https://user-images.githubusercontent.com/19331824/185580181-f9a422f1-8732-44e6-8d56-241121467b2e.png">

JustinBeBoy commented on ui/modal/text_input_modal.go:
```diff
@@ -86,17 +86,17 @@ func (tm *TextInputModal) SetTextWithTemplate(template string) *TextInputModal {
 
 func (tm *TextInputModal) Handle() {
 
 	if editorsNotEmpty(tm.textInput.Editor) {
```
it should be `tm. tm.isEnabled = editorsNotEmpty(tm.textInput.Editor)`

Sirmorrison commented:
* add notice when sign message success like mockup and remove background of "signature" text
![image](https://user-images.githubusercontent.com/27733432/185968523-0425f5e8-1371-40bd-a24e-605e68f93758.png)
There is a Pr https://github.com/planetdecred/godcr/pull/1039 already to resolve this no need implementing this twice.

Sirmorrison committed:
- Rebase Pr
- Set fixed modal width.
- Fix lint error from PR rebase.
- Update verify message info modal message to mockup
- Update subpage info modal button color and position.

Sirmorrison committed:
Fix truncated amount dcr unit bug.
- Update create wallet account modal to updated mockup design
- Minor code clean up

JustinBeBoy commented on ui/page/wallet_settings_page.go:
```diff
@@ -65,39 +100,67 @@ func NewWalletSettingsPage(l *load.Load, wal *dcrlibwallet.Wallet) *WalletSettin
 				return pg.pageContainer.Layout(gtx, len(w), func(gtx C, i int) D {
-					return w[i](gtx)
+					return layout.Inset{Left: values.MarginPadding50}.Layout(gtx, w[i])
+					// return w[i](gtx)
```
remove it

JustinBeBoy commented on ui/page/wallet_settings_page.go:
```diff
@@ -370,83 +574,180 @@ func (pg *WalletSettingsPage) HandleUserInteractions() {
+		NegativeButtonStyle(pg.Theme.Color.Primary, pg.Theme.Color.Surface).
+		PositiveButton(values.String(values.StrResetDexClient), func(isChecked bool) bool {
+			if pg.Dexc().Reset() {
+				pg.Toast.Notify("DEX client data reset complete.")
```
localizable

JustinBeBoy commented on ui/page/wallet_settings_page.go:
```diff
@@ -370,83 +574,180 @@ func (pg *WalletSettingsPage) HandleUserInteractions() {
+			if pg.Dexc().Reset() {
+				pg.Toast.Notify("DEX client data reset complete.")
+			} else {
+				pg.Toast.NotifyError("DEX client data reset failed. Check the logs.")
```
localizable

JustinBeBoy commented:
1 nothing show when change account name successful
<img width="800" alt="1111" src="https://user-images.githubusercontent.com/19331824/186201928-e1131760-37b1-427a-ac0b-318897a15764.png">

2. please remove back button
<img width="801" alt="2222" src="https://user-images.githubusercontent.com/19331824/186201957-c877058d-5b17-4009-bf39-d0290bb24fcc.png">

Sirmorrison committed:
make settings page a main page, fix rename success toast
